community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -366,14 +365,15 @@ public Object nodeGetProperty( long nodeId, int propertyKeyId ) throws EntityNot
     }
 
     @Override
-    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int... relTypes )
+    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int[] relTypes )
             throws EntityNotFoundException
     {
         statement.assertOpen();
         try ( Cursor<NodeItem> node = dataRead().nodeCursorById( statement, nodeId ) )
         {
-            return new CursorRelationshipIterator(
-                    node.get().relationships( direction( direction ), deduplicate( relTypes ) ) );
+            return new CursorRelationshipIterator( dataRead()
+                    .nodeGetRelationships( statement, node.get(), direction( direction ),
+                            asSet( relTypes, t -> t >= 0 ) ) );
```

Comment:
We were silently filtering negative ids before (I am not sure if on purpose or not): https://github.com/neo4j/neo4j/pull/8792/files#diff-35e966ebb9c89256d6078481e6170e09L865 (without the filtering there are tests failures). I am happy to reconsider that, but I think this PR is not the right place for that.

I'll have a look at the performance, but mind that it might be slower to build the set, but the actual check if a type is in the set or not, could be faster than scanning the array. So we might need to look at the bigger picture.

It didn't look to me that we were taking any advantage of the fact that the array was sorted. E.g., https://github.com/neo4j/neo4j/pull/8792/files#diff-962abdeb972f25d4dc93b35378804fdeL232 Am I missing something?